misc/server/src/behaviour.rs

```diff
@@ -0,0 +1,78 @@
+use libp2p::autonat;
+use libp2p::identify;
+use libp2p::kad::{record::store::MemoryStore, Kademlia, KademliaConfig};
+use libp2p::ping;
+use libp2p::relay;
+use libp2p::swarm::behaviour::toggle::Toggle;
+use libp2p::{identity, swarm::NetworkBehaviour, Multiaddr, PeerId};
+use std::str::FromStr;
+use std::time::Duration;
+
+const BOOTNODES: [&str; 4] = [
+    "QmNnooDu7bfjPFoTZYxMNLWUQJyrVwtbZg5gBMjTezGAJN",
+    "QmQCU2EcMqAqQPR2i9bChDtGNJchTbq5TbXJJ16u19uLTa",
+    "QmbLHAnMoJPWSCR5Zhtx6BHJX9KiKNN6tpvbUcqanj75Nb",
+    "QmcZf59bWwK5XFi76CZX8cbJ4BhTzzA3gU1ZjYZcYW3dwt",
+];
+
+#[derive(NetworkBehaviour)]
+pub(crate) struct Behaviour {
+    relay: relay::Behaviour,
+    ping: ping::Behaviour,
+    identify: identify::Behaviour,
+    pub(crate) kademlia: Toggle<Kademlia<MemoryStore>>,
+    autonat: Toggle<autonat::Behaviour>,
+}
+
+impl Behaviour {
+    pub(crate) fn new(
+        pub_key: identity::PublicKey,
+        enable_kademlia: bool,
+        enable_autonat: bool,
```

Comment:
Why do we need this config?

Reply:
As in, why do we not enable them always?

In the case of the rust-libp2p IPFS bootstrap node, I don't have enough experience with our `libp2p-autonat` implementation as to run it on a critical piece of infrastructure.

Does that reasoning make sense @thomaseizinger?

Reply:
Should we just remove autonat from here then? Somewhere else you said that you consider this server to be production-ready. If so, we should be able to just unconditionally enable these protocols or remove them.

We have plenty of examples for how to add more protocols to an implementation. Users can easily fork this and add more if they want. If we are not using this for the bootstrap node, then I'd vote for removing it.

Reply:
I still see value in it. Once we have https://github.com/libp2p/rust-libp2p/issues/4226 I would like to first enable AutoNAT on the rust-libp2p based IPFS **staging** bootstrap node and later on the **production** one. Thus I would re-introduce the flag.

Do you feel strongly about it @thomaseizinger? Given that `libp2p-server` is versioned separately to `libp2p` we can remove it any day.

Reply:
Don't feel strongly, just wanted to understand why it is there :)